gemrb/core/Strings/StringView.h

```diff
@@ -110,14 +110,22 @@ class StringViewImp {
 		return *(data + i);
 	}
 
+	bool operator==(const StringViewImp<CharT>& other) const noexcept {
+		if (other.length() != length()) {
+			return false;
+		}
+
+		return std::char_traits<CharT>::compare(data, other.data, 0) == 0;
```

Comment:
this `0` is a mistake, no?

Reply:
also I believe `compare`  returns 0 when they are equal

Reply:
Yeah, that's something I was running over as well. [Docs ](https://en.cppreference.com/w/cpp/string/char_traits/compare) say:
> Compares the first `count` characters of the character strings `s1` and `s2`. The comparison is done lexicographically.
If `count` is zero, strings are considered equal. 

I am assuming _equal_ as _equal length_. I can check this later again but for what other reason would it reasonably be designed this way.

Reply:
its just saying if you pass a 0, no checking is done. you should pass one of the lengths.

Reply:
This is probably a good opportunity to add a test that compares two strings of the same length and different chars :)

Reply:
Indeed ...